tests/common/test_event.py

```diff
@@ -19,23 +19,50 @@
 
 import json
 import os
-import threading
 from datetime import datetime, timedelta
 
 from azurelinuxagent.common import event, logger
-from azurelinuxagent.common.event import add_event, elapsed_milliseconds, EventLogger, report_metric, WALAEventOperation
+from azurelinuxagent.common.event import add_event, elapsed_milliseconds, EventLogger, report_metric, \
+    WALAEventOperation, parse_xml_event, parse_json_event, parse_event
 from azurelinuxagent.common.exception import EventError
 from azurelinuxagent.common.future import OrderedDict, ustr
 from azurelinuxagent.common.protocol.wire import GoalState
-from azurelinuxagent.common.telemetryevent import TelemetryEventParam
+from azurelinuxagent.common.telemetryevent import TelemetryEvent, TelemetryEventParam
 from azurelinuxagent.common.utils import fileutil
 from azurelinuxagent.common.utils.extensionprocessutil import read_output
-from azurelinuxagent.common.version import CURRENT_AGENT, CURRENT_VERSION
-from azurelinuxagent.ga.monitor import MonitorHandler
-from tests.tools import AgentTestCase, data_dir, load_data, Mock, patch
+from azurelinuxagent.common.version import AGENT_NAME, CURRENT_AGENT, CURRENT_VERSION
+from tests.ga.test_monitor import SysInfoMock
+from tests.tools import AgentTestCase, data_dir, load_data, Mock, patch, PropertyMock
 
 
 class TestEvent(AgentTestCase):
+
+    def setUp(self):
+        AgentTestCase.setUp(self)
+
+        self.event_dir = self.tmp_dir
+        event.init_event_logger(self.event_dir)
+
+        self.mock_sysinfo = patch("azurelinuxagent.common.sysinfo.SysInfo.get_instance", return_value=SysInfoMock)
+        self.mock_sysinfo.start()
+
+    def tearDown(self):
+        self.mock_sysinfo.stop()
+
+    def test_parse_xml_event(self, *args):
+        data_str = load_data('ext/event_from_extension.xml')
+        event = parse_xml_event(data_str)
+        self.assertNotEqual(None, event)
```

Comment:
since we are here, can we improve test_parse_* to do more substantial checks?

Reply:
I'd really prefer doing it on a separate, smaller PR. I can add a task for myself.